### ISSUE TITLE
Initial route update doesn't happen if ngView in a template loaded by ngInclude

### ISSUE THREAD

**Issue by jswright** [angular/angular.js#1213]
I can't seem to reproduce this on jsFiddle (mhevery says it's because you need real async with delay to reproduce, but the only way I know to include templates for ngInclude in jsFiddle are inline script tags), but here's how to set it up with two local files:

### index.html
```html
<html>
  <head>
    <script src="http://docs.angularjs.org/angular-1.0.1.min.js"></script>
    <script type="text/javascript">
      angular.module('myApp', [], function($routeProvider) {
        $routeProvider.when('/topic', {
          controller: TopicsCtrl,
          template: 'Controller: {{name}}'
        });
        $routeProvider.when('/topic/:topicId', {
          controller: TopicCtrl,
          template: 'Controller: {{name}}<br />Topic ID: {{params.topicId}}'
        });
        $routeProvider.otherwise({redirectTo: '/topic'});
      });

      function TopicsCtrl($scope) {
        $scope.name = 'TopicsCtrl';
      };

      function TopicCtrl($scope) {
        $scope.name = 'TopicCtrl';
        $scope.params = $routeParams;
      };
    </script>
  </head>
  <body ng-app="myApp">
    <div ng-include src="'app.html'"></div>
  </body>
</html>
```

### app.html
```html
<a href="#/topic">All topics</a>
<a href="#/topic/1">Topic 1</a>
<a href="#/topic/2">Topic 2</a>
<a href="#/topic/3">Topic 3</a>
<hr />
<div ng-view></div>
```

## Expected results:
Page loads, redirects to #/topic, and shows 'Controller: TopicCtrl' in the view.

## Actual results:
Page loads, does not redirect.  If you click any link, the proper view is loaded.  However, if you then reload, the view will not be properly loaded.

If you move the ng-view into `index.html` it works fine.

**Comment by luckylooke** [comment]
ng 1.3.8, reload fix worked for me :)

**Comment by wcjr** [comment]
I believe this is also causing me to have to do some silly stuff in my unit tests to workaround it:  http://stackoverflow.com/questions/29170970/why-does-route-have-to-be-included-in-unit-testing-location-only

**Comment by DutchProgrammer** [comment]
This bug still exists in  1.3.15

by adding the following code it's fixed
app.run(['$route', function($route)  {
  $route.reload();
}]);

**Comment by DutchProgrammer** [comment]
This bug still exists in  1.3.15

by adding the following code it's fixed
app.run(['$route', function($route)  {
  $route.reload();
}]);

**Comment by blackbeltdev** [comment]
The bug still exists in 1.4.1 as well. We have deeply nested ng-include that has a 'page' directive which contains ng-view in the template.

It seems just the act of injecting $route into the run block does the trick for us. Someone else pointed that out earlier in thread, "tobyn commented on Sep 6, 2013" but the noop isn't necessary just the injection.

```javascript
// doesn't work!
app.run(['$log', '$rootScope', function ($log, $rootScope) { 
  // nothing
}
```

```javascript
// works!
app.run(['$log', '$rootScope', '$route', function ($log, $rootScope, $route) { 
  // nothing
}
```

**Comment by mstroming** [comment]
+1 to blackbeltdev's comment.

**Comment by blackbeltdev** [comment]
There's an interesting sentence in here which might explain why injecting the $route fixes the problem with ng-include.

http://www.bennadel.com/blog/2770-route-must-be-injected-in-order-to-enable-the-routechangesuccess-event-in-angularjs.htm

"Instead, I have another, optional, sub-controller which does nothing but inject the $route service (which forces AngularJS to call the $routeProvider)."

**Comment by davidsandoz** [comment]
The fix was reverted in https://github.com/angular/angular.js/commit/75bf80700e992e6d60f16f6360d400ad28049d36

Then I suppose the issue should be re-opened.

**Comment by Narretz** [comment]
> It will be re-applied for `v1.6.x`, with a breaking change notice and possibly a way to disable
the feature is tests.

We just have to make sure the BC notice is added to the CL when the time comes.

**Comment by goddabuzz** [comment]
+1 to blackbeltdev's comment. =) Works for 1.5.x

**Comment by HaroldStruwig** [comment]
this is still not fixed in 1.5.8

**Comment by gkalpak** [comment]
@HaroldStruwig, this was fixed on master with #14893 (and will be included in the upcoming v1.6.0 release. The fix can't be backported as-is, due to a slight breaking change.

Although we could backport it, but change the default to to non-eager `$route` instantiation (to avoid the breaking change), it isn't worth it. Manually fixing this is easier than opting-into eager-instantiation. E.g.:

```js
// IF we had backported the fix (which we won't)
app.config(['$routeProvider', function ($routeProvider) {
  $routeProvider.eagerInstantiationEnabled(true);
}]);

// vs

// Manually fixing it in v1.5.x (much sorter and recommended)
app.run(['$route', angular.noop]);
```

--
The work-around/fix for v1.5.x is even document [here][1].

[1]: https://code.angularjs.org/1.5.8/docs/api/ngRoute/directive/ngView#known-issues
